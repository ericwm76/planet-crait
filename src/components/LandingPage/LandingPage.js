import React, { Component } from 'react';
import './LandingPage.scss';

class LandingPage extends Component {
    constructor() {
      super();
      this.state = {
        name: '',
        quote: ''
      }
    }

    render() {
        return(
            <section className='landing-page'>
                 <h2>Welcome to Planet Crait</h2>
              <form className='landing-form'>
                  <input className='name-input' 
                         type='text'
                         placeholder='Name of Jedi'
                         name='name'
                         value={this.state.name} />
                  <input className='quote-input' 
                         type='text'
                         placeholder='Your favourite Star Wars quote here'
                         name='quote'
                         value={this.state.title} />
                  <div className='fandom-level-btns'>
                    <button className='novice-btn'>Novice</button>
                    <button className='intermediate-btn'>Intermediate</button>
                    <button className='Expertnovice-btn'>Expert</button>
                  </div>
 
                  <button className='submit-btn' >Do. Or do not. Submit</button>
              </form>
            </section>
        )
    }

}

export default LandingPage;