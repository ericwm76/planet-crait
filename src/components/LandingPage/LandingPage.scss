.landing-page {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  height: 100vh;
  background-image: url('https://media.giphy.com/media/nEfs1tRuKsXxm/giphy.gif');
  background-repeat: no-repeat;
  background-size: cover;
}

.landing-form {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 40%;
}

.name-input,
.quote-input {
  font-size: 2em;
  border: none;
  border-radius: 5px;
  margin: 5px;
  width: 110%;
}

.submit-btn {
  font-size: 2em;
  border: none;
  border-radius: 5px;
  margin: 10px;
  padding: 5px;
}

