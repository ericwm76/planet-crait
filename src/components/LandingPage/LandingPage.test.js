import React from 'react';
import { shallow } from 'enzyme';
import LandingPage from './LandingPage';

describe('LandingPage', () => {
  let wrapper;
  const mockSetUsers = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<LandingPage setUsers={mockSetUsers}/>)
  })

  it('should match snapshot with all data passing in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should update state when input is filled', () => {

     const updateState = {
       
     }
  })
}) 