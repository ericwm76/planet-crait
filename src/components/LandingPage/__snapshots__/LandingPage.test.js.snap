// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should match snapshot with all data passing in correctly 1`] = `
<section
  className="landing-page"
>
  <div
    className="form-section"
  >
    <h2
      className="welcome-crait"
    >
      Welcome to Planet Crait
    </h2>
    <form
      className="landing-form"
    >
      <input
        className="name-input"
        name="name"
        onChange={[Function]}
        placeholder="Name of Jedi"
        type="text"
        value=""
      />
      <input
        className="quote-input"
        name="quote"
        onChange={[Function]}
        placeholder="Your favourite Star Wars quote here"
        type="text"
      />
      <div
        className="fandom-level-btns"
      >
        <button
          className="novice-btn btn"
          name="level"
          onClick={[Function]}
          value="Novice"
        >
          Novice
        </button>
        <button
          className="intermediate-btn btn"
          name="level"
          onClick={[Function]}
          value="Intermediate"
        >
          Intermediate
        </button>
        <button
          className="expert-btn btn"
          name="level"
          onClick={[Function]}
          value="Expert"
        >
          Expert
        </button>
      </div>
      <Link
        className="submit-btn"
        onClick={[Function]}
        to="/movies"
      >
        Do. Or do not. Submit
      </Link>
    </form>
  </div>
</section>
`;
